pilot/pkg/security/authn/utils/utils.go

```diff
@@ -69,11 +71,30 @@ func BuildInboundTLS(mTLSMode model.MutualTLSMode, node *model.Proxy,
 
 	// Set Minimum TLS version to match the default client version and allowed strong cipher suites for sidecars.
 	ctx.CommonTlsContext.TlsParams = &tls.TlsParameters{
-		TlsMinimumProtocolVersion: tls.TlsParameters_TLSv1_2,
-		CipherSuites:              SupportedCiphers,
+		CipherSuites: SupportedCiphers,
 	}
-
+	// Configure TLS version based on meshconfig TLS API.
+	// The min TLS version is initialized to be at least TLS 1.2.
+	minTLSVersion := meshconfig.MeshConfig_TLSConfig_TLSV1_2
+	if mc != nil && mc.GetMeshMTLS() != nil {
+		minTLSVersion = mc.GetMeshMTLS().MinProtocolVersion
+	}
+	ctx.CommonTlsContext.TlsParams.TlsMinimumProtocolVersion,
+		ctx.CommonTlsContext.TlsParams.TlsMaximumProtocolVersion = getTLSVersions(minTLSVersion)
 	authn_model.ApplyToCommonTLSContext(ctx.CommonTlsContext, node, []string{}, /*subjectAltNames*/
 		trustDomainAliases, ctx.RequireClientCertificate.Value)
 	return ctx
 }
+
+func getTLSVersions(mcMinTLSVer meshconfig.MeshConfig_TLSConfig_TLSProtocol) (tls.TlsParameters_TlsProtocol, tls.TlsParameters_TlsProtocol) {
+	// The min TLS version is at least TLS 1.2.
+	minVersion := tls.TlsParameters_TLSv1_2
+	// If user configures the min TLS version to be TLS 1.3, the min TLS version will be TLS 1.3.
+	switch mcMinTLSVer {
+	case meshconfig.MeshConfig_TLSConfig_TLSV1_3:
+		minVersion = tls.TlsParameters_TLSv1_3
+	}
+	// The max TLS version is configured as TLS 1.3
+	maxVersion := tls.TlsParameters_TLSv1_3
+	return minVersion, maxVersion
+}
```

Comment:
```suggestion

	ctx.CommonTlsContext.TlsParams.TlsMinimumProtocolVersion,
		ctx.CommonTlsContext.TlsParams.TlsMaximumProtocolVersion = getTLSVersions(mc.GetMeshMTLS().GetMinProtocolVersion())
	authn_model.ApplyToCommonTLSContext(ctx.CommonTlsContext, node, []string{}, /*subjectAltNames*/
		trustDomainAliases, ctx.RequireClientCertificate.Value)
	return ctx
}

func getTLSVersions(ver meshconfig.MeshConfig_TLSConfig_TLSProtocol) (tls.TlsParameters_TlsProtocol, tls.TlsParameters_TlsProtocol) {
	switch ver {
	case meshconfig.MeshConfig_TLSConfig_TLSV1_3:
		return tls.TlsParameters_TLSv1_3, tls.TlsParameters_TLSv1_3
	default:
		return tls.TlsParameters_TLSv1_2, tls.TlsParameters_TLSv1_3
	}
}
```

Can simplify this a lot

Reply:
The PR has been revised to simplify getTLSVersions(). The nil checkings of mc != nil && mc.GetMeshMTLS() != nil are needed because mc and mc.GetMeshMTLS() may be nil. For example, mc or mc.GetMeshMTLS() are nil in some existing tests and they crash without the nil checkings.

Reply:
`GetMeshMTLS` (and any protobuf Get) handles nil

Reply:
The getTLSVersions() in the comment failed at lint test with the following unparam error. The lint error is due to the second return parameter is always tls.TlsParameters_TLSv1_3. To fix the lint error, the original getTLSVersions() is used.

{pilot/pkg/security/authn/utils/utils.go:89:102: getTLSVersions - result 1 (github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/v3.TlsParameters_TlsProtocol) is always 4  : getTLSVersions - result 1 (github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/v3.TlsParameters_TlsProtocol) is always 4
Category: unparam
File: pilot/pkg/security/authn/utils/utils.go
Line: 89

Reply:
Why not just not have the function return the max tls version?

All the code needs to do is get the mesh setting or a default - there is no need to make it 50 lines of complex code

Reply:
Agreed. I have revised the function based on the comments.